pkg/resource/container_image.go

```diff
@@ -0,0 +1,426 @@
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return err
+		}
+		path, err := sanitizePath(path, header.Name)
+		if err != nil {
+			logger.Error("%v - skipped", err)
+			continue
+		}
+		info := header.FileInfo()
+		if info.IsDir() {
+			if err = os.MkdirAll(path, 0700); err != nil {
+				return err
+			}
+			continue
+		}
+
+		file, err := os.OpenFile(path, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0600) // #nosec G304
+		if err != nil {
+			logger.Error("%v", err)
+			continue
+		}
+		defer file.Close()
+
+		// Copy a maximum number of bytes (layer size * 10) so we do not get "bombs". It is unlikely that a file
+		// with significant entropy will be compressed more than 10x. We can review this.
+		n, err := io.CopyN(file, tarReader, size)
+		if err != nil && err != io.EOF {
+			return fmt.Errorf("could not copy file to disk: %v", err)
+		}
+		if n >= size {
+			logger.Warning("copying file %s did not finish due to max file size: %v", file.Name(), err)
+		}
+	}
+	return nil
+}
+
+// ClonePath returns where this repo has been cloned if cloned else ""
+func (r *ContainerImage) ClonePath() string {
+	return r.clonePath
+}
+
+// Depth returns the depth for things that have version control
+func (r *ContainerImage) Depth() uint16 {
+	return 0
+}
+
+// EnrichResult adds contextual information to each result
+func (r *ContainerImage) EnrichResult(result *response.Result) *response.Result {
+
+	hash, file, found := strings.Cut(result.Location.Path, string(os.PathSeparator))
+	if found {
+		result.Location.Version = hash
+		result.Location.Path = file
+	}
+	result.Notes = r.labels
+	if result.Location.Version != "" {
+		// If there is no layer then it is a metadata file
+		result.Kind = response.ContainerMetdataResultKind
+	}
+	result.Kind = response.ContainerLayerResultKind
+
+	result.Contact.Name, result.Contact.Email = r.Contact()
+
+	return result
+}
+
+// SetDepth allows you to adjust the depth for the resource
+func (r *ContainerImage) SetDepth(depth uint16) {
+	// no-op
+}
+
+// SetCloneTimeout lets you adjust the timeout before the clone aborts
+func (r *ContainerImage) SetCloneTimeout(timeout time.Duration) {
+	// no-op
+}
+
+// Since returns the date after which things should be scanned for things
+// that have versions
+func (r *ContainerImage) Since() string {
+	return ""
+}
+
+// ReadFile provides a way to access values in the resource
+func (r *ContainerImage) ReadFile(path string) ([]byte, error) {
+	return os.ReadFile(filepath.Join(r.ClonePath(), filepath.Clean(path)))
+}
+
+// Walk traverses the resource like a directory tree
+func (r *ContainerImage) Walk(fn WalkFunc) error {
+	// Handle if path is a file
+	if fs.FileExists(r.ClonePath()) {
+		file, err := os.Open(r.ClonePath())
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+
+		// path is empty because it's not in a directory
+		return fn("", file)
+	}
+
+	return filepath.WalkDir(r.ClonePath(), func(path string, d iofs.DirEntry, err error) error {
```

Comment:
I wonder if we could reuse the Files resource here a bit.

Maybe during the clone above we could create a Files resource and save it on as something like `r.extractedFiles`

and this could just become:

```go
return r.extractedFiles.Walk(walkFunc)
```

Or if we make the manifest and configs a JSONData objects:

```go
if err := r.manifest.Walk(walkFunk); err != nil {
	return err
}

if err := r.ociConfig.Walk(walkFunk); err != nil {
	return err
}

return r.extractedFiles.Walk(walkFunc)
```

or something like that.